### ISSUE TITLE
Too many log options in helm --help output.

### ISSUE THREAD

**Issue by cdrage** [helm/helm#7163]
Output of `helm version`: N/A
Output of `kubectl version`: N/A
Cloud Provider/Platform (AKS, GKE, Minikube etc.): N/A

See the current global flags:

```sh
Global Flags:
      --add-dir-header                   If true, adds the file directory to the header
      --alsologtostderr                  log to standard error as well as files
      --debug                            enable verbose output
      --kube-context string              name of the kubeconfig context to use
      --kubeconfig string                path to the kubeconfig file
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --log-file string                  If non-empty, use this log file
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
      --logtostderr                      log to standard error instead of files (default true)
  -n, --namespace string                 namespace scope for this request
      --registry-config string           path to the registry config file (default "/home/wikus/.config/helm/registry.json")
      --repository-cache string          path to the file containing cached repository indexes (default "/home/wikus/.cache/helm/repository")
      --repository-config string         path to the file containing repository names and URLs (default "/home/wikus/.config/helm/repositories.yaml")
      --skip-headers                     If true, avoid header prefixes in the log messages
      --skip-log-headers                 If true, avoid headers when opening log files
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
  -v, --v Level                          number for the log level verbosity
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
```

Many of these options are seldom used, especially related to:

```sh
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --log-file string                  If non-empty, use this log file
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
```

IMO, this should be truncated into a more standarized / minimal output:

```sh
Usage:
  helm package [CHART_PATH] [...] [flags]

Flags:
      --app-version string       set the appVersion on the chart to this version
  -u, --dependency-update        update dependencies from "Chart.yaml" to dir "charts/" before packaging
  -d, --destination string       location to write the chart. (default ".")
  -h, --help                     help for package
      --key string               name of the key to use when signing. Used if --sign is true
      --keyring string           location of a public keyring (default "/home/wikus/.gnupg/pubring.gpg")
      --set stringArray          set values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)
      --set-file stringArray     set values from respective files specified via the command line (can specify multiple or separate values with commas: key1=path1,key2=path2)
      --set-string stringArray   set STRING values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)
      --sign                     use a PGP private key to sign this package
  -f, --values strings           specify values in a YAML file or a URL(can specify multiple)
      --version string           set the version on the chart to this semver version

Global Flags:
      --add-dir-header                   If true, adds the file directory to the header
      --debug                            enable verbose output
      --kube-context string              name of the kubeconfig context to use
      --kubeconfig string                path to the kubeconfig file
  -n, --namespace string                 namespace scope for this request
      --registry-config string           path to the registry config file (default "/home/wikus/.config/helm/registry.json")
      --repository-cache string          path to the file containing cached repository indexes (default "/home/wikus/.cache/helm/repository")
      --repository-config string         path to the file containing repository names and URLs (default "/home/wikus/.config/helm/repositories.yaml")

Log Flags:
      --alsologtostderr                  log to standard error as well as files
      --skip-headers                     If true, avoid header prefixes in the log messages
      --skip-log-headers                 If true, avoid headers when opening log files
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
  -v, --v Level                          number for the log level verbosity
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --log-file string                  If non-empty, use this log file
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
      --logtostderr                      log to standard error instead of files (default true)
```

This can be easily setup using a custom template file with cobra13.

Alternatively, the extra logging parameters could be disabled from these lines of code: https://github.com/helm/helm/blob/f1dc84773f9a34fe59a504fdb32428ce1d56a2e8/cmd/helm/helm.go#L59

What do you think @mattfarina ? Would make the parameters a lot less daunting IMO. I can push a PR for this, but thought to setup a minor discussion before doing so.

**Comment by mattfarina** [comment]
I like the change to more groupings. It makes the flags easier to skim. We cannot disable them as it would break backwards compatibility and we should not remove them from display as people need to learn about them.

I would like to see a PR for this.

**Comment by cdrage** [comment]
@mattfarina I'm on it :)

**Comment by rjminchuk** [comment]
Is there any documentation on these logging options? I'm trying to understand how these work/where helm might log to, and I haven't found anything substantial. I'm happy to read through the source code to try and make sense of it if someone could point me where to look.

**Comment by marckhouzam** [comment]
This is really nice!

To continue the discussion along the lines of hiding the logging flags: every help message still 'dumps' a lot of text for global/logging flags, and users must scroll up to read the actual help for the command in question.

We could go the route `kubectl` has gone, and extract the global flag list into its own command.
So `kubectl <anything> -h` does not show the global flags but ends the help with the sentence:
```
Use "kubectl options" for a list of global command-line options (applies to all commands).
```
Personally, I think that helm has few enough **global** flags that we can keep them in the help messages, but we could remove the **logging** flags and have a command like `helm options` (to be consistant with `kubectl`) which would list both **global** and **logging** flags.

**Comment by tillig** [comment]
I just now tried to figure out log verbosity and failed horribly. It seems you have to "know" that klog is being used, and the description `-v, --v Level     number for the log level verbosity` assumes you know what the number should be. However, doing a `helm upgrade` with... any number there - `helm upgrade -v 0` or `helm upgrade -v 5` or whatever... doesn't seem to yield any different output. So it's not really possible to tell if these have any effect or what. Better docs here would definitely help.

**Comment by bacongobbler** [comment]
For reference, a verbosity level greater than 10 returns HTTP request output. I use `-v 10`  often to check and see what requests Helm is invoking against Kubernetes.

**Comment by cdrage** [comment]
Anyone has my permission to use my code here: https://github.com/helm/helm/pull/7178 update it with @bacongobbler 's review and push it.

Unfortunately I don't have much time to work on it. But anyone is free to take it up!

**Comment by marckhouzam** [comment]
This was actually addressed by #8067 which removed the klog flags from the help output.  It is available starting with helm 3.4
